src/spi_bus.cpp

```diff
@@ -0,0 +1,392 @@
+            case MotorBoardCommand::ENABLE_MTR1:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.motor1->Enable() : udriver.motor1->Disable();
+                break;
+            case MotorBoardCommand::ENABLE_MTR2:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.motor2->Enable() : udriver.motor2->Disable();
+                break;
+            case MotorBoardCommand::ENABLE_VSPRING1:
+                break;
+            case MotorBoardCommand::ENABLE_VSPRING2:
+                break;
+            case MotorBoardCommand::SEND_CURRENT:
+                break;
+            case MotorBoardCommand::SEND_POSITION:
+                break;
+            case MotorBoardCommand::SEND_VELOCITY:
+                break;
+            case MotorBoardCommand::SEND_ADC6:
+                break;
+            case MotorBoardCommand::SEND_ENC_INDEX:
+                break;
+            case MotorBoardCommand::SEND_ALL:
+                break;
+            case MotorBoardCommand::SET_CAN_RECV_TIMEOUT:
+                break;
+            case MotorBoardCommand::ENABLE_POS_ROLLOVER_ERROR:
+                command_content == MotorBoardCommand::ENABLE ?
+                    udriver.EnablePositionRolloverError() :
+                    udriver.DisablePositionRolloverError();
+                break;        
+            default:
+                throw std::runtime_error("SpiBus::send_if_input_changed:"
+                                        " Error cannot send unkown command");
+                break;
+        } // end switch
+    } // end for
+}
+
+void SpiBus::send_newest_controls()
+{
+    // send controls -----------------------------------------------------------
+    for(size_t i = 0; i < control_.size(); ++i)
+    {
+        if(control_[i]->length() == 0)
+        {
+            rt_printf("You tried to send control on the motor[%ld] "
+                      "but no control has been set\n", i);
+            exit(-1);
+        }
+        MotorInterface::ScalarTimeseries::Index timeindex =
+            control_[i]->newest_timeindex();
+
+        double control = (*control_[i])[timeindex];
+        
+        main_board_interface_->motors[i].SetCurrentReference(control);
+        control_[i]->tag(timeindex);
+        sent_control_[i]->append(control);
+    }
+    motors_are_paused_ = false;
+    main_board_interface_->SendCommand();
+}
+
+bool SpiBus::is_ready()
+{
+    // check if the motors and if the udriver are enabled and ready.
+    bool is_hardware_ready = true;
+    for(size_t i = 0 ; i < nb_udrivers_ ; ++i)
+    {
+        MotorDriver& udriver = main_board_interface_->motor_drivers[i];
+        is_hardware_ready = is_hardware_ready &&
+                            udriver.motor1->IsEnabled() &&
+                            udriver.motor2->IsEnabled() &&
+                            udriver.motor1->IsReady() &&
+                            udriver.motor2->IsReady();
```

Comment:
Does this mean, there is no option to only run a single motor on a udriver when using the master board?

Reply:
I think it is also the case with the can bus no?